table/manifest_operations_test.go

```diff
@@ -0,0 +1,514 @@
+		// 1. Identify all snapshots still referenced by table metadata
+		// 2. Collect all manifests referenced by those snapshots
+		// 3. Find manifests on disk not referenced by any snapshot
+		// 4. Safely delete unreferenced manifests
+
+		// For this test, we verify the basic structure is correct
+		s.T().Log("Manifest cleanup test completed - verified manifest accessibility")
+		s.T().Log("Real cleanup would require catalog-level snapshot management")
+	})
+
+	s.Run("SafetyChecksForCleanup", func() {
+		// Test safety checks to prevent deleting referenced manifests
+		ident := table.Identifier{"default", "cleanup_safety_test"}
+		tbl, _ := s.createTableWithManyFiles(ident, 2, false)
+
+		// Get current manifests
+		snapshot := tbl.CurrentSnapshot()
+		s.Require().NotNil(snapshot)
+
+		fs := s.getFS(tbl)
+		manifests, err := snapshot.Manifests(fs)
+		s.Require().NoError(err)
+
+		// Verify all manifests are properly referenced
+		for _, manifest := range manifests {
+			// Each manifest should be readable and have valid structure
+			entries, err := manifest.FetchEntries(fs, false)
+			s.Require().NoError(err)
+
+			// Verify manifest contains expected entries
+			s.Greater(len(entries), 0, "Referenced manifest should have entries")
+
+			// Verify manifest metadata
+			s.NotEmpty(manifest.FilePath(), "Manifest should have valid path")
+			s.Greater(manifest.Length(), int64(0), "Manifest should have non-zero length")
+
+			s.T().Logf("Manifest %s: %d entries, %d bytes",
+				manifest.FilePath(), len(entries), manifest.Length())
+		}
+
+		// In a real cleanup implementation, these checks would prevent deletion:
+		// 1. Manifest is referenced by current snapshot ✓
+		// 2. Manifest is referenced by other snapshots (check metadata history)
+		// 3. Manifest file is not corrupted ✓
+		// 4. Manifest contains valid entries ✓
+
+		s.T().Log("Safety checks passed - all manifests are properly referenced and valid")
+	})
+}
+
+func (s *ManifestOperationsTestSuite) TestManifestMetrics() {
+	s.Run("ManifestSizeAndCount", func() {
+		// Test manifest size and count metrics
+
+		// Test with different manifest merge settings
+		testCases := []struct {
+			name           string
+			mergeEnabled   bool
+			minCount       string
+			targetSize     string
+			expectedResult string
+		}{
+			{
+				name:           "No Merging",
+				mergeEnabled:   false,
+				minCount:       "100",
+				targetSize:     "8388608", // 8MB
+				expectedResult: "Multiple small manifests",
+			},
+			{
+				name:           "Aggressive Merging",
+				mergeEnabled:   true,
+				minCount:       "1",
+				targetSize:     "1024", // 1KB - very small for testing
+				expectedResult: "Fewer, larger manifests",
+			},
+			{
+				name:           "Conservative Merging",
+				mergeEnabled:   true,
+				minCount:       "10",
+				targetSize:     "8388608", // 8MB
+				expectedResult: "Limited merging",
+			},
+		}
+
+		for i, tc := range testCases {
+			s.Run(tc.name, func() {
```

Comment:
this is the proper time you would use `s.Run` :smile: for subtests that share code between them